### PR TITLE
feat: add Learning page content

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes

Adds tabs for private content, public content and syllabi on the learning page. If private content is empty, the tab is hidden. The public content will query all courses with 'qiskit' origin.

Related #263 
Resolves #200 

## Implementation details

- Use carbon-web-components tab elements to show tabs
- Placeholder empty private courses array–the private content tab is hidden if there are not private courses

## How to read this PR
[Preview link](https://saiba-pr-269.tfb1j0wd7x2.us-south.codeengine.appdomain.cloud/account/learning)

## Screenshots
Screenshot 2022-11-03 at 11 31 22 AM

### CODE REVIEW

File: services/course.ts [hunk 1]
```diff
@@ -41,3 +41,42 @@ export const getCourse = (courseId: string, lang = 'en-US') => {
     (res: Query<CourseListDTO>) => res.data.courses[0]
   );
 };
+
+export const getPublicCourses = (language = 'en-US') => {
+  const query = /* GraphQL */ `
+    query GetPublicCourses($language: String) {
+      courses(filter: { origin: { _eq: "qiskit" } }) {
```
Comment (abdonrd) [inline services/course.ts:48]: I imagine that this filter should be forced on the API side for the user role.

File: store/course/utils.ts [hunk 2]
```diff
@@ -21,6 +22,7 @@ export function courseDTOToCourse(courseDTO: CourseDTO): Course {
     slug,
     description,
     units: units.map(({ units_id }) => units_id.id),
+    unitsObj: units.map(({ units_id }) => unitDTOToUnit(units_id)),
```
Comment (abdonrd) [inline store/course/utils.ts:25]: Why this? I see in the previous line that the units are already being saved.

File: store/course/store.ts [hunk 3]
```diff
@@ -99,8 +100,17 @@ export const useCourseStore = defineStore('courseStore', () => {
     );
   }
 
+  function fetchPublicCourses() {
+    return useLazyAsyncData('fetch-public-courses', () => {
+      return getPublicCourses(lang.value).then(
+        (data) => (publicCourses.value = data.map(courseDTOToCourse))
```
Comment (abdonrd) [inline store/course/store.ts:106]: I miss something like this to store the units in the units store:

https://github.com/Qiskit/saiba/blob/44bbff150f1191ab3206ed1b5999029a03b7ac76/store/course/store.ts#L85-L89

File: services/course.ts (hunk 1 above)
Comment (yoshiemuranaka) [reply services/course.ts:48]: Won't I potentially need to query private courses with this endpoint?

File: store/course/utils.ts (hunk 2 above)
Comment (yoshiemuranaka) [reply store/course/utils.ts:25]: the units prop is only storing the unit ID string. I was considering refactoring the code that is using the units prop to use unitsObj instead, but wanted to keep this PR smaller–let me know if you think these changes should be included

File: store/course/store.ts (hunk 3 above)
Comment (yoshiemuranaka) [reply store/course/store.ts:106]: Let me know if you want me to make these changes in this PR!

File: services/course.ts (hunk 1 above)
Comment (yoshiemuranaka) [reply services/course.ts:48]: Maybe I misunderstood, can the backend enforce visible courses depending on whether or not they are added as members of a cohort?

Comment (yoshiemuranaka) [reply services/course.ts]: Resolved by https://github.com/Qiskit/saiba/pull/269/commits/14efd780e8cd1ad7aa8b1ca279369874895e00f8

File: store/course/store.ts (hunk 3 above)
Comment (yoshiemuranaka) [reply store/course/store.ts]: resolved by https://github.com/Qiskit/saiba/pull/269/commits/8de643b1a6ad451250e9a786e94feba31a6b0776

File: store/course/utils.ts (hunk 2 above)
Comment (yoshiemuranaka) [reply store/course/utils.ts]: resolved by https://github.com/Qiskit/saiba/pull/269/commits/02a50081aeafcd47cbd9595b914c4085ec33513d

File: components/learning/PublicContent.vue [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    <p class="public-course__title">{{ course.title }}</p>
+    <div class="public-course__units-list">
+      <div
+        v-for="unit of course.units"
+        :key="unit"
+        class="public-course__units-item"
+      >
+        <ProgressCircle :progress="0" />
+        <BasicLink
+          :to="`/course/${course.slug}/${unitStore.unitById(unit).value?.slug}`"
```
Comment (yoshiemuranaka) [inline components/learning/PublicContent.vue:18]: optional chaining required to avoid es lint error `Object is possibly 'undefined'.`

**Comment by yoshiemuranaka** [comment]
This PR is ready for review

File: store/course/store.ts [hunk 5]
```diff
@@ -53,6 +54,10 @@ export const useCourseStore = defineStore('courseStore', () => {
     () => activeCourseUnits.value?.[activeUnitIndex.value - 1]
   );
 
+  const publicCourses = computed(() =>
+    learnerCourses.value.filter(({ origin }) => origin === Origin.QISKIT)
+  );
```
Comment (abdonrd) [inline store/course/store.ts:59]: What do you think to move the origin logic to the query itself?
I'm thinking when we need to paginate we should not request both courses (public and private) mixed in the same query.

Comment (yoshiemuranaka) [reply store/course/store.ts]: Resolved by https://github.com/Qiskit/saiba/pull/269/commits/3a3ffdf3ae323a40f36ce352df6d281cfe823303

**Review by abdonrd — APPROVED** [review]
👏🏻

**Review by abdonrd — APPROVED** [review]

**Pull request merged**